adidas/lakehouse-engine
# [Suggestion] - Mkdocs Material as the base for documentation

Thank you everyone for the great project. Apologies if this PR looks big but in fact is just the initial setup. Let me explain what is this and how it does work.

I'm an opensource contributor and the creator of [Esmerald framework](https://esmerald.dev). I absolutely love your project and I thought I could help you with some PR and contribute to the growth and health of the codebase (if allowed, of course).

n the last few years, [Mkdocs Material](https://squidfunk.github.io/mkdocs-material/) became almost a *must have* for documentation of these type of projects. I know a lot of people also prefer `sphinx` which is very mature and amazing as well, I just find `mkdocs-material` easier to setup.

Besides Esmerald and some other opensource projects I have built, there are a lot of known projects also using this package for their documentation, for instance, [Pydantic](https://pydantic.dev/), [FastAPI](https://fastapi.tiangolo.com/), [Mkdocstrings](https://mkdocstrings.github.io/python/) and list goes on and on.

I though I could initially suggest for this PR a move to this tool and from there it would be a lot easier to grow and maintain the documentation.

I had the pleasure of talking with one of your team members about this and I thought it would be nice to suggest and see how it goes. [Mkdocs Material](https://squidfunk.github.io/mkdocs-material/).

Deploying it it is extremely easy  and even maintaining a consistent API Documentation won't be so hard since Python has some defaults that can help you with that to automatically generate those for you. Example of an automatic generated API Documentation [would be like this one](https://esmerald.dev/references/esmerald/).

## How to test this

I added some extra `make commands` to make it easier for everyone who wants to contribute, to be able to do it easily like running documentation locally, installing requirements without headaches...

```shell
make requirements-local
```

or

```
scripts/install
```

Will make sure it will install all the requirements in your local machine (including for docs).

```shell
scripts/docs
```

Will start the documentation locally and by default serving on `https://localhost:8000`.

## How to add documentation with mkdocs-material

Well, as you can see from the PR, if you want to add a new file, for example a page called `filters`, you only need to do this:

1. Inside the `docs/` create a file called `filters.md`.
2. Add the `filters.md` to `mkdocs.yml` under the `nav:` section and that's it!

Of course if you want to create documentation sections, the docs of mkdocs explain well but it would be like (inside mkdocs.yml):

```yaml
...
nav:
   - New Section Name:
      - "filters.md" 
```

## What is this `scripts` folder

As per README, it follows a practice ["Scripts to Rule Them All"](https://github.com/github/scripts-to-rule-them-all).

## How to deploy

I could see your docs are published using github actions, so it is already 99% done and mkdocs material actually gives you a [clear example](https://squidfunk.github.io/mkdocs-material/publishing-your-site/?h=deploy#with-github-actions) how to do it quicky.

Something around this line `- run: mkdocs gh-deploy --force`.
